C2CX API Integration

nomanurrehman opened the pull request:
This pull request addresses issue 663.

I have not yet added the specs as I want to first confirm if the implementation is correct. Once it has been confirmed, I will be happy to add the required tests.

estherleongym commented on lib/cryptoexchange/exchanges/c2cx/services/pairs.rb:
```diff
@@ -0,0 +1,29 @@
+              market: C2cx::Market::NAME
+            )
+          end
+          pairs
```
name pairs `market_pairs` instead for consistency.

nomanurrehman commented:
@estherleongym I have changed the variable name and also updated some of the API calls while I was writing the tests as I found a few bugs.

Please review the last commits and let me know if there are any issues.

Thanks.

wongy91 commented on lib/cryptoexchange/exchanges/c2cx/services/market.rb:
```diff
@@ -0,0 +1,42 @@
+          ticker.bid       = NumericHelper.to_d(data['buy'])
+          ticker.ask       = NumericHelper.to_d(data['sell'])         
+          ticker.volume    = NumericHelper.divide(NumericHelper.to_d(data['volume']), ticker.last)
+          ticker.timestamp = Time.now.to_i
```
please use data['timestamp'], eg
ticker.timestamp = data['timestamp'].to_i

wongy91 commented on lib/cryptoexchange/exchanges/c2cx/services/pairs.rb:
```diff
@@ -0,0 +1,29 @@
+        def adapt(response)
+          market_pairs = []
+          response['data']['SupportPairs'].each do |pair|
+            base, target = pair.split('_')
```
`https://api.c2cx.com/v1/ticker/?symbol=BTC_ETH` returns

```
{
    "code": 200,
    "message": "success",
    "data": {
        "timestamp": "1532052144",
        "high": 0.0651,
        "last": 0.0622,
        "low": 0.0618,
        "buy": 0.0622,
        "sell": 0.0624,
        "volume": 339.0794601449
    }
}
```

the base, target is inverted so,
`target, base` = pair.split('_')

wongy91 commented on spec/exchanges/c2cx/integration/market_spec.rb:
```diff
@@ -0,0 +1,110 @@
+    expect(hashed_pairs).to include({base: 'USDT', target: 'XRP', market: market})
+    expect(hashed_pairs).to include({base: 'USDT', target: 'EOS', market: market})
+    expect(hashed_pairs).to include({base: 'USDT', target: 'WAX', market: market})
+    expect(hashed_pairs).to include({base: 'USDT', target: 'XLM', market: market})
```
good idea trying to test every pair however it is not necessary as exchanges add/remove pairs all the time.

nomanurrehman commented:
@estherleongym @wongy91 All changes have been made.
